core/src/main/scala/chisel3/DataImpl.scala

```diff
@@ -470,21 +470,32 @@ private[chisel3] trait DataImpl extends HasId with NamedComponent { self: Data =
     }
   }
 
+  // Specializes the .toString method of a [[Data]] for conditions such as
+  //  DataView, Probe modifiers, a DontCare, and whether it is bound or a pure chisel type
   private[chisel3] def stringAccessor(chiselType: String): String = {
+    // Add probe and layer color (if they exist) to the returned String
+    def addProbeModifier(chiselType: String): String = {
+      probeInfo match {
+        case None => chiselType
+        case Some(ProbeInfo(writeable, layer)) =>
+          val layerString = layer.map(x => s", ${x.fullName}").getOrElse("")
+          (if (writeable) "RWProbe" else "Probe") + s"<$chiselType$layerString>"
+      }
+    }
     // Trace views to give better error messages
     // Reifying involves checking against ViewParent which requires being in a Builder context
     // Since we're just printing a String, suppress such errors and use this object
     val thiz = Try(reifySingleTarget(this)).toOption.flatten.getOrElse(this)
     thiz.topBindingOpt match {
-      case None => chiselType
+      case None => addProbeModifier(chiselType)
       // Handle DontCares specially as they are "literal-like" but not actually literals
       case Some(DontCareBinding()) => s"$chiselType(DontCare)"
       case Some(topBinding) =>
         val binding: String = thiz._bindingToString(topBinding)
         val name = thiz.earlyName
```

Comment:
I'm sort of surprised the test is below is working:
```scala
Wire(probe.Probe(UInt(1.W))).toString should be("BoundDataModule.?: Wire[Probe<UInt<1>>]")
```
A Wire of probe should have a top binding...

Reply:
It does have a top binding? Not sure I understand..

Reply:
I think I missed that you also had `addProbeModifier` below.

I think it would be better to only do that once, will suggest above.